### PR TITLE
fix(#20): Add `params` option to the mustermann grape pattern for taking into account the parameter type

### PR DESCRIPTION
**Pull request by jcagarcia**
Hey 👋 

I've decided to create this PR for including the `params` options to the mustermann grape pattern. Fixing #20.

I've added several tests and seems that if we provide the proper params information, we are able to generate more precise regular expressions.

```ruby
Mustermann.new('/:id', type: :grape, params: {"id"=>{:type=>"Integer"}}).params('/1') # => { id: '1'}
Mustermann.new('/:id', type: :grape, params: {"id"=>{:type=>"Integer"}}).params('/foo') # => nil
Mustermann.new('/:id', type: :grape, params: {"id"=>{:type=>"String"}}).params('/foo') # => { id: 'foo'}
```

What do you think?

### CODE REVIEW

**Review by dblock — COMMENTED** [review]
I like it! Thanks @jcagarcia.

Let's increment the version to 1.1.0 here since this is a new feature.

@namusyaka, you around?

**Comment by dblock** [comment]
As you fix CI, see if it's easy to add a Ruby 3.x, feel free to make separate PRs.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+### 1.0.3 (Next)
```
[the remark this answers is not in the document]
Comment (jcagarcia) [reply CHANGELOG.md]: Totally welcome!! 👍

**Comment by jcagarcia** [comment]
Hi @dblock ,

Thanks a lot for the review. I've updated the version to `1.1.0`. Also, I've added Ruby 3.x to the gitHub action that runs the tests.

Let me know if something else is needed! :)

**Comment by dblock** [comment]
Will let this marinate a little and see if we hear from @namusyaka, otherwise I'll merge soon.

**Comment by jcagarcia** [comment]
Hi again @dblock 👋 

Do you think we can merge this change and release `1.1.0` version of `mustermann-grape`?

**Comment by dblock** [comment]
I released 1.1.0.

**Comment by dblock** [comment]
@jcagarcia Want to help us with maintaining mustermann-grape? LMK your rubygems username if you're down.

**Comment by jcagarcia** [comment]
> @jcagarcia Want to help us with maintaining mustermann-grape? LMK your rubygems username if you're down.

Sure @dblock ! Really happy to help :) This is my rubygems profile https://rubygems.org/profiles/jcagarcia

**Comment by dblock** [comment]
Accept the maintainer invite here: https://github.com/ruby-grape/mustermann-grape/invitations. Added you to Rubygems! Thanks.

**Pull request merged**